plugins/server/plugin.go

```diff
@@ -0,0 +1,111 @@
+package clientport
+
+import (
+	"encoding/binary"
+	"errors"
+	"net"
+
+	"github.com/coredhcp/coredhcp/handler"
+	"github.com/coredhcp/coredhcp/logger"
+	"github.com/coredhcp/coredhcp/plugins"
+	"github.com/insomniacslk/dhcp/dhcpv4"
+	"github.com/insomniacslk/dhcp/dhcpv6"
+)
+
+var log = logger.GetLogger()
+
+func init() {
+	plugins.RegisterPlugin("server", setupServerV6, setupServerV4)
+}
+
+// StaticRecords holds a MAC -> IP address mapping
+var StaticRecords map[string]net.IP
+
+// DHCPv6Records and DHCPv4Records are mappings between MAC addresses in
+// form of a string, to network configurations.
+var (
+	serverIP     net.IP
+	netmask      net.IP
+	ClientSubnet net.IPMask
+)
+
+// Handler6 not implemented only IPv4
+func Handler6(req, resp dhcpv6.DHCPv6) (dhcpv6.DHCPv6, bool) {
+	return resp, true
+}
+
+// Handler4 handles DHCPv4 packets for the file plugin
+func Handler4(req, resp *dhcpv4.DHCPv4) (*dhcpv4.DHCPv4, bool) {
+	if req.OpCode != dhcpv4.OpcodeBootRequest {
+		log.Printf("plugins/server: Not a BootRequest!")
+	}
+	switch mt := req.MessageType(); mt {
+	case dhcpv4.MessageTypeDiscover:
```

Comment:
I wouldn't have a server plugin. I would completely remove this.

* The type of the reply message should be moved to the `MainHandler`: https://github.com/coredhcp/coredhcp/blob/master/coredhcp.go#L165

* The subnet mask can be set in the file plugin since there we define a subnet (or it can be a separate plugin).

* The server identifier can be set in the main handler if you want to automatically get the local IP address of where CoreDHCP is running or if you want to make it configurable you can use the `server_id` plugin

* The router can be a separate plugin if you want to make it configurable or it can be set in the file plugin if you want to use the 1st address of the subnet defined there.